File: backend/models/Product.js
import mongoose from "mongoose";

const productSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    price: { type: Number, required: true },
    image: { type: String }, // link cloudinary
    description: { type: String },
  },
  { timestamps: true }
);

export default mongoose.model("Product", productSchema);

File: le_nhut_cuong_chieut5/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final double price;
  final String? description;
  final String? image;

  Product({
    required this.id,
    required this.name,
    required this.price,
    this.description,
    this.image,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['_id'],
      name: json['name'],
      price: (json['price'] as num).toDouble(),
      description: json['description'],
      image: json['image'],
    );
  }
}
